Backend Changes (Community API Update for litmus-portal).

ishangupta-ds opened the pull request:
What this PR does / why we need it:

Community API Update for litmus-portal.
Adds month of year wise data with formatted end-dates.

gdsoumya commented:
There are changes made to the front-end components most probably linting problem. cc: @arkajyotiMukherjee

arkajyotiMukherjee commented:
The front-end changes are still showing up in files changes @gdsoumya 
Shouldn't it be just one file analytics.go?

ishangupta-ds commented:
> There are changes made to the front-end components most probably linting problem. cc: @arkajyotiMukherjee

Actually i removed all extra components i added and updated the code for frontend from master branch thus changes are showing up but the code is as it was earlier, the frontend remains the same. @gdsoumya @arkajyotiMukherjee

ishangupta-ds commented:
> The front-end changes are still showing up in files changes @gdsoumya
> Shouldn't it be just one file analytics.go?

The changes were made on top of the yaml editor updates where the components were removed and all affected components were replaced with the code from master iself, you can check the preview for verifying. @gdsoumya @arkajyotiMukherjee @rajdas98

chandankumar4 commented on app/client/package.json:
```diff
@@ -1,63 +1,64 @@
+	"name": "create-react-app-typescript-redux",
+	"version": "2.0.0",
+	"dependencies": {
+		"@material-ui/core": "4.9.0",
```
Why this indentation changed?

ishangupta-ds replied:
This is happening due to the vs code configuration.

chandankumar4 replied:
ok! This vs code configuration, every developer doesn't have, So then they will do the change a/c to their configuration, Then how would you handle that?

arkajyotiMukherjee replied:
We have to add a linting setup similar to litmus portal to fix this too @chandankumar4

arkajyotiMukherjee replied:
I'll take that on me....after the release

ishangupta-ds replied:
Yeah as we don't have any linting aetup here for the style to be consistent for th tab spaces i can change it to 2 from 4.

arkajyotiMukherjee replied:
for now you can chnage your editor to take two space for a tab and run npm run format @ishangupta-ds

ishangupta-ds replied:
I have updated the indentation from 4 to 2 tab space.

chandankumar4 replied:
What about the other file change in front-end?

arkajyotiMukherjee replied:
@ishangupta-ds you edited the wrong file with wrong configurations...now there is 30 files change....just add one file in this PR that's the analytics.go
leave all other files

arkajyotiMukherjee replied:
don't put any ts, tsx, js, jsx, json files or any other frontend related files in the PR

chandankumar4 commented on app/client/src/components/InstallChaos/index.tsx:
```diff
@@ -1,57 +1,57 @@
-	description: string;
-	yamlLink: string;
-	engine: boolean;
+	title: string
```
Why the semicolon(;) is removed?

ishangupta-ds replied:
I am using an auto formatter so it removes any unnecessary code.

arkajyotiMukherjee replied:
What formatter are you using...we used prettier and the code was consistent with semicolons everywhere...If you are using prettier use the prettierrc provided in src folder as the config file for the extension or just enable semicolons in the vscode prettier settings itself @ishangupta-ds

ishangupta-ds replied:
I was also using prettier.

ishangupta-ds replied:
I have updated the rc file from 4 to 2 spaces tab width and also enabled semi-colon.

chandankumar4 commented on app/server/pkg/community/google/analytics.go:
```diff
@@ -105,6 +108,47 @@ func getTotalCounts() error {
 }
 
+//timeSeriesHelper is a helper function to add month-end dates to all monthly data
+func timeSeriesHelper(TimeData [][]string) ([][]string, error) {
```
Is there any npm library for this? So that we don't have to make do this by ourselves

ishangupta-ds replied:
This could have been done on frontend but the computation will unnecessarily increase whereas on the backend we are caching the data , also this same api will be required for Repute thus  have kept it in backend.

chandankumar4 replied:
ok

chandankumar4 commented on app/server/pkg/community/google/analytics.go:
```diff
@@ -143,5 +189,30 @@ func getGraphData() error {
+	}
+	monthlyOperatorData := response.Rows
+
+	monthlyOperatorDataCorrected, err := timeSeriesHelper(monthlyOperatorData)
```
```suggestion
	monthlyOperatorDataCorrected, err := timeSeriesHelper(response.Rows)
```

chandankumar4 replied:
```suggestion
	formattedMonthlyOperatorData, err := timeSeriesHelper(monthlyOperatorData)
```

chandankumar4 commented on app/server/pkg/community/google/analytics.go:
```diff
@@ -143,5 +189,30 @@ func getGraphData() error {
+	if err != nil {
+		return fmt.Errorf("Error while getting response, err: %s", err)
+	}
+	monthlyOperatorData := response.Rows
```
```suggestion
```

chandankumar4 commented on app/server/pkg/community/google/analytics.go:
```diff
@@ -143,5 +189,30 @@ func getGraphData() error {
+	}
+	monthlyExperimentData := response.Rows
+
+	monthlyExperimentDataCorrected, err := timeSeriesHelper(monthlyExperimentData)
```
Same comment as above

ishangupta-ds replied:
okay i am on it.

ishangupta-ds commented:
@chandankumar4 the suggested changes have been made.

ishangupta-ds commented:
@gdsoumya @arkajyotiMukherjee @chandankumar4 the formatting was done as required.

arkajyotiMukherjee commented:
@ishangupta-ds 
This might be helpful
https://stackoverflow.com/questions/39459467/remove-a-modified-file-from-pull-request

ishangupta-ds commented:
> can you only add the analytics.go file in the PR
> we don't need the other files....don't do a `git add .` use `git add analytics.go` instead

You asked me to format it right?

ishangupta-ds commented:
> can you only add the analytics.go file in the PR
> we don't need the other files....don't do a `git add .` use `git add analytics.go` instead

I will have to roll back to a previous commit and fix many conflicting files after that as we require the items for release it may be needed that we do it ASAP, the formatting can be improved later on.

ishangupta-ds commented:
> https://stackoverflow.com/questions/39459467/remove-a-modified-file-from-pull-request

I will have to remove 20+ files from staging one by one and resolve conflicts it will delay the work for charthub and litmus-portal as well.

rajdas98 commented:
/preview

rajdas98 commented:
Removed the other frontend files from this PR.
